images/kindnetd/cmd/kindnetd/main.go

```diff
@@ -134,27 +179,17 @@ func main() {
 }
 
 // nodeNodesReconciler returns a reconciliation func for nodes
-func makeNodesReconciler(cniConfig *CNIConfigWriter, hostIP string) func(*corev1.NodeList) error {
+func makeNodesReconciler(cniConfig *CNIConfigWriter, hostIP string, ipFamily IPFamily, clientset *kubernetes.Clientset) func(*corev1.NodeList) error {
 	// reconciles a node
 	reconcileNode := func(node corev1.Node) error {
-		// first get this node's IP
-		nodeIP := internalIP(node)
-		if nodeIP == "" {
-			klog.Infof("Node %v has no Internal IP, ignoring\n", node.Name)
-			return nil
-		}
-
-		// don't do anything unless there is a PodCIDR
-		podCIDR := node.Spec.PodCIDR
-		if podCIDR == "" {
-			klog.Infof("Node %v has no CIDR, ignoring\n", node.Name)
-			return nil
-		}
-
+		// first get this node's IPs
+		// we don't support more than one IP address per IP family for simplification
+		nodeIPs := internalIPs(node)
+		klog.Infof("Handling node with IPs: %v\n", nodeIPs)
 		// This is our node. We don't need to add routes, but we might need to
-		// update the cni config.
-		if nodeIP == hostIP {
-			klog.Infof("handling current node\n")
+		// update the cni config and "annotate" our external IPs
```

Comment:
this is no longer needed, just forget to update the comment, but we no longer need to annotate IPs
This is going ready to merge soon :smile: @BenTheElder

Reply:
is it? commeent still mentions annotation